packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -329,6 +329,31 @@ export class RunningSummarizer implements IDisposable {
         this.initialized = true;
     }
 
+    /**
+     * Blocks a new summarizer from running in case RefreshSummaryAck is being processed.
+     * Assumes that caller checked upfront for lack of concurrent action (this.refreshSummaryAckLock)
+     * before calling this API. I.e. caller is responsible for either erroring out or waiting on this promise.
+     * Note: The refreshSummaryAckLock makes sure no summarizer gets enqueued or processed
+     * until the refresh has completed. One can't rely uniquely on the summarizingLock as the
+     * refreshLatestSummaryAck also happens during the time summarizingLock !== undefined.
+     * Ex. Summarizer submits a summay + op and then waits for the Summary Ack to proceed
+     * with the refreshLatestSummaryAck and complete the summary.
+     * @param action - action to perform.
+     * @returns - result of action.
+     */
+    public async lockedRefreshSummaryAckAction<T>(action: () => Promise<T>) {
```

Comment:
So we want to allow refresh to happen from within summary, thus we are not checking that summary is in flight.
But how do we ensure that refresh can't happen outside of summary process, while summary is in flight?
I believe right now it's not possible simply because we are not processing any ops while summary is in flight, and thus we will not process summaryAck as well. On top of that I've added asserts that validate that latest summary knowledge does not change through summary process (excluding refresh at the beginning). Feels like that's sufficient but wanted to point out that in theory we might allow unwanted activity.

Using multi-threading analogy, we would use single lock (i.e. not add new locks) and allow reentrancy (of taking lock by same thread). The problem in JS - there is no implicit context (like thread ID) so we do not know this call is done in context of another call.

Just sharing, not sure if we can do anything here. :)